include/nccl_ofi_ep_addr_list.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NCCL_OFI_EP_ADDR_LIST_H
+#define NCCL_OFI_EP_ADDR_LIST_H
+
+#include "nccl_ofi.h"
+
+struct nccl_ofi_ep_addr_list;
+typedef struct nccl_ofi_ep_addr_list nccl_ofi_ep_addr_list_t;
+
+/**
+ * Initialize an endpoint-address-set pair list
+ *
+ * NOTE: there is no destroy function because this list is stored with device,
+ * which has no destructor.
```

Comment:
Do some cleanup in the new device finalize API (even if it won't be used by NCCL).